adr/0017-agent-operations-framework.md

```diff
@@ -0,0 +1,129 @@
+# 17. Agent Operations Framework
+
+Date: 2025-01-27
+
+## Status
+
+Accepted
+
+## Context
+
+Following the new epics and product decisions, we need to build a framework capable of executing write operations on SAP machines.
+To achieve this, we have decided to use the agent, as it runs on the machines with root privileges, has all the necessary permissions to execute commands, and includes infrastructure to consume messages from other components of Trento.
+
+Unlike fact-gathering operations, we have chosen to delegate this new development to a dedicated library rather than embedding the code directly into the agent's codebase. This approach ensures better separation of concerns and allows the release of additional tools supporting operations without affecting the agent's codebase.
+
+The requirements for this new development are very specific:
+
+- Operations must be atomic and include rollback capabilities.
+- Operations must accept arguments to enable actions in different contexts without requiring dedicated code for each case.
+- Operations must be transactional, including distinct steps for prerequisites verification, commit, rollback, and validation of changes applied during the commit phase.
+- Operations must be versioned, with backward compatibility ensured for previous versions in the event of updates.
+- Operations must be idempotent.
+
+The agent will consume the library containing these operations and use it to fulfill operation requests from other components.
+
+## Decision
+
+The library handling operations is named [workbench](https://github.com/trento-project/workbench).
+
+## Glossary
```

Comment:
Honestly, I don't really like this `Glossary` chapter. It is rather confusing and I think the subsequent next chapters perfectly explain all the details.
I don't think we need such a summary.
I would personally remove it